### PR TITLE
add initial implementation of resevervation get cmd

### PR DESCRIPTION
**Pull request by David-M-IBM**
Created for [issue #414](https://zenhub.ibm.com/workspaces/tech-zone-accelerator-toolkit-627a5df0f8dae200104ff923/issues/skol/ocp-installer/414)

Usage: reservation get --reservation-id 63fcb95d9991b60017f119ea 

Output looks like:
image

### CODE REVIEW

File: cmd/getReservation.go [hunk 1]
```diff
@@ -1,21 +1,66 @@
...
 	Short: "Gets a specific reservation.",
 	Long:  `Get the details of a reservation.`,
-	Run: func(cmd *cobra.Command, args []string) {
+	RunE: func(cmd *cobra.Command, args []string) error {
 		logger.Debug("Getting your reservation...")
+		return getReservation(cmd, args)
 	},
 }
 
+func getReservation(cmd *cobra.Command, args []string) error {
```
Comment (nathanagood) [inline cmd/getReservation.go:28]: 👍

File: cmd/getReservation.go [hunk 2]
```diff
@@ -1,21 +1,66 @@
...
+var reservationID string
+
 // getReservationCmd represents the viewReservation command
 var getReservationCmd = &cobra.Command{
 	Use:   "get",
 	Short: "Gets a specific reservation.",
 	Long:  `Get the details of a reservation.`,
-	Run: func(cmd *cobra.Command, args []string) {
+	RunE: func(cmd *cobra.Command, args []string) error {
 		logger.Debug("Getting your reservation...")
```
Comment (nathanagood) [inline cmd/getReservation.go:23]: Consider using `Debugf` and printing out the `reservationId` right here.

**Review by nathanagood — CHANGES_REQUESTED** [review]
A couple minor points in the review here. The template changes would make the code

File: pkg/reservations/base.go [hunk 3]
```diff
@@ -88,6 +92,33 @@ func (w *TextWriter) Write(out io.Writer, rez TZReservation) error {
 	return nil
 }
 
+func (w *TextWriter) WriteSingleRes(out io.Writer, rez TZReservation) error {
```
[the remark this answers is not in the document]
Comment (David-M-IBM) [reply pkg/reservations/base.go]: Kk updated this in latest commit.

File: pkg/reservations/base.go [hunk 4]
```diff
@@ -88,6 +92,33 @@ func (w *TextWriter) Write(out io.Writer, rez TZReservation) error {
...
+	// TODO: Probably get this from a resource file of some kind
+	consoleTemplate := ` - {{.Name}} - {{.Status}}
+   Reservation Id: {{.ReservationId}}
+   Description: {{.Description}}
+   Collection Id: {{.CollectionId}}
+   Extend Count: {{.ExtendCount}}
+   Service Links:
+    --------------------------------`
+
+	for _, serviceLink := range rez.ServiceLinks {
```
[the remark this answers is not in the document]
Comment (David-M-IBM) [reply pkg/reservations/base.go]: I added this in the latest commit. Definitely think it looks a lot cleaner.

File: pkg/reservations/base.go [hunk 5]
```diff
@@ -88,6 +91,30 @@ func (w *TextWriter) Write(out io.Writer, rez TZReservation) error {
...
+	consoleTemplate := ` - {{.Name}} - {{.Status}}
+   Reservation Id: {{.ReservationId}}
+   Description: {{.Description}}
+   Collection Id: {{.CollectionId}}
+   Extend Count: {{.ExtendCount}}
+   Service Links:
+    --------------------------------
+    {{- range .ServiceLinks}}
+		{{- if .Sensitive}}
+			{{- printf "\n    %s: ****Private****\n    --------------------------------" .Label}}
```
Comment (nathanagood) [inline pkg/reservations/base.go:105]: Nice!

**Review by nathanagood — APPROVED** [review]
LGTM; nice changes.

**Pull request merged**